### PR TITLE
Implement downloader

### PR DESCRIPTION
**Pull request by marcorosa**
Implementation of a downloader (as an external file and class).
Once the PR is merged, we can consider integrating it with the Client

### CODE REVIEW

**Comment by marcorosa** [comment]
Also the downloader, like the uploader, uses functions from the `FileCrypto` class. 
So it is better to take it out from `uploader.py`

**Comment by steenzout** [comment]
@marcorosa conflicting files ^.

**Comment by marcorosa** [comment]
@steenzout  I have a lot of print function. The plan was to implement a downloader, make a PR with it and then do some code refactoring to replace all the print funcitons with loggers (and then do a second PR).
But if you think that loggers should be implemented before this PR, I'll integrate them now.

**Comment by steenzout** [comment]
@marcorosa let's do it now.

File: storj/sharder.py [hunk 1]
```diff
@@ -0,0 +1,136 @@
+import math
+import os
+
+# global SHARD_MULTIPLES_BACK, MAX_SHARD_SIZE
```
Comment (marcorosa) [inline storj/sharder.py]: solved

File: storj/file_crypto.py [hunk 2]
```diff
@@ -0,0 +1,59 @@
+from Crypto.Cipher import AES
+from Crypto import Random
+import hashlib
+
+
+class FileCrypto:
+
+    def encrypt_file(self, algorithm, file_path, encrypted_file_save_path, password):
+        if algorithm == "AES":
```
Comment (marcorosa) [inline storj/file_crypto.py]: solved

File: storj/file_crypto.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            with open(file_path, 'rb') as in_file, open(encrypted_file_save_path, 'wb') as out_file:
+                self.encrypt_file_aes(in_file, out_file, password)
+
+    def decrypt_file(self, algorithm, file_path, decrypted_file_save_path, password):
+        if algorithm == "AES":
```
Comment (marcorosa) [inline storj/file_crypto.py]: solved

File: storj/file_crypto.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        while not finished:
+            chunk, next_chunk = next_chunk, cipher.decrypt(in_file.read(1024 * bs))
+            if len(next_chunk) == 0:
+                padding_length = ord(chunk[-1])
+                if padding_length < 1 or padding_length > bs:
+                    raise ValueError("bad decrypt pad (%d)" % padding_length)
+                # all the pad-bytes must be the same
+                if chunk[-padding_length:] != (padding_length * chr(padding_length)):
+                    # this is similar to the bad decrypt:evp_enc.c from openssl program
+                    raise ValueError("bad decrypt")
```
Comment (marcorosa) [inline storj/file_crypto.py]: solved. why do you prefer single quotes?

**Comment by marcorosa** [comment]
Everything should be ok now.
Does pep8 want a blank line at the end of files? I see opposing opinions about it

File: storj/downloader.py [hunk 5]
```diff
@@ -36,15 +38,16 @@ def set_file_metadata(self, bucket_id, file_id):
             file_metadata = self.client.file_metadata(bucket_id, file_id)
             # Get file name
             self.filename_from_bridge = str(file_metadata.filename)
-            print "Filename from bridge: " + self.filename_from_bridge
+            self.__logger.debug("Filename from bridge: " +
```
Comment (steenzout) [inline storj/downloader.py]: in this case you can do:

```
self.__logger.debug("Filename from bridge: %s", self.filename_from_bridge)
```

the format and values will then be replaced by the `logging` library if `debug` is enabled.

this differs from the cases where you just want to build a string.

for example:
```
msg = 'Filename from bridge: %s' % self.filename_from_bridge
```

_remember, single quotes_ :)

**Comment by marcorosa** [comment]
With the last commit I've changed the occurrences of `+`s with `%s`

**Pull request merged**